lib/App/Critique/Command/collect.pm

```diff
@@ -42,14 +42,7 @@ sub execute {
 
     my $filter;
     if ( my $f = $opt->filter ) {
-        if ( ref $f eq 'CODE' ) {
-            $filter = $f;
-        }
-        else {
-            $filter = $opt->invert
-                ? sub { $_[0]->stringify !~ /$f/ }
-                : sub { $_[0]->stringify =~ /$f/ };
-        }
+        $filter = file_filter($opt,'file_filter_regex');
     }
```

Comment:
Hmm, okay I see where you are going here, however I really would like to avoid the `no strict refs` thing and have it be a hard reference to the actual subroutine (`\&file_filter_regex`).